monoio/src/buf/io_vec_buf.rs

```diff
@@ -162,7 +162,7 @@ pub unsafe trait IoVecBufMut: Unpin + 'static {
     unsafe fn set_init(&mut self, pos: usize);
 }
 
-unsafe impl IoVecBufMut for VecBuf {
+unsafe impl<B: IoBufMut + IoBuf> IoVecBufMut for VecBuf<B> {
```

Comment:
here we are assuming for simplicity that all `B: IoBufMut` also has `B: IoBuf`. If not, we can also decouple.